libraries/tlv-account-resolution/src/account.rs

```diff
@@ -61,6 +115,23 @@ impl ExtraAccountMeta {
             is_writable: is_writable.into(),
         })
     }
+
+    /// Resolve an `ExtraAccountMeta` into an `AccountMeta`, potentially
+    /// resolving a program-derived address (PDA) if necessary
+    pub fn resolve(&self, instruction: &Instruction) -> Result<AccountMeta, ProgramError> {
+        let mut account_meta = if self.discriminator == 1 {
+            let seeds = Seed::unpack_address_config(&self.address_config)?;
+            AccountMeta {
+                pubkey: resolve_pda(&seeds, instruction)?,
+                is_signer: self.is_signer.into(),
+                is_writable: self.is_writable.into(),
+            }
+        } else {
+            AccountMeta::try_from(self)?
+        };
+        de_escalate_account_meta(&mut account_meta, &instruction.accounts);
+        Ok(account_meta)
+    }
```

Comment:
Is there a reason that this changed from a match statement to an if statement? The match statement is more exhaustive, so I'd prefer to stick with that.

Reply:
It can definitely be a match statement!

In #4963 I had changed it to `if meta.is_pda()`, and when you do `AccountMeta::try_from(self)?` it will error on any discriminators outside of `0`, so I figured cases were covered.

But happy to convert to match